### ISSUE TITLE
Extend with MLPClassifer

### ISSUE THREAD

**Issue by Wilsonlou** [automl/auto-sklearn#525]
Dear experts,

I tried to extend MLP from sklearn as a candidate model. However,  it seems very hard to set search space for hidden_layer_sizes as a categorical hyper, for example:
hidden_layer_sizes = CategoricalHyperparameter(
            "hidden_layer_sizes", [(20,10),(10,10)]
        )

I tried OrdinalHyperparameter, but seems auto-sklearn still not supports it. 

Appreciate if you can provide any help on it.

Thx

Wilson

**Comment by mfeurer** [comment]
@ahn1340 were you able to progress on the extending Auto-sklearn example?

**Comment by Wilsonlou** [comment]
Thx guys, actually the issue here is the hyperparam is a list of tuples, for other params, it's ok to work with.

**Comment by Wilsonlou** [comment]
Btw, I tried params like: Shuffle, early_stopping, nesterovs_momentum, these params are true or false. Once I tried to set like:
        nesterovs_momentum = CategoricalHyperparameter(
          'nesterovs_momentum', [0, 1], default_value=1
        )
        early_stopping = CategoricalHyperparameter(
           'early_stopping', [0,1], default_value=0
        )

It will show value error due to value type check in sklearn MLP code:
    def _validate_hyperparameters(self):
        if not isinstance(self.shuffle, bool):
            raise ValueError("shuffle must be either True or False, got %s." %
                             self.shuffle)
https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/neural_network/multilayer_perceptron.py

**Comment by Wilsonlou** [comment]
For these params are true or false:
I tried the below yesterday, but due to the hyperparam validate function in MLP code, it will failed as it is string not bool, so I tried 1,0 and "True","False" and True, False, none of them worked.

nesterovs_momentum = CategoricalHyperparameter(
'nesterovs_momentum', ["True", "False"], default_value="True"
)
For hyperparam of tuples, will have a try.

Thx

**Comment by Wilsonlou** [comment]
For hyperparam of tuples, have tried your suggestion, it works, thx:)

The issue here is Boolean params now, is there anyway to get rid of that hyperparam validation fuction?

Some piece of source code from MLPClassifier:
def _validate_hyperparameters(self):
**if not isinstance(self.shuffle, bool):**
raise ValueError("shuffle must be either True or False, got %s." %
self.shuffle)

**Comment by mfeurer** [comment]
Closing this due to inactivity - also, there's now an example on this: http://automl.github.io/auto-sklearn/master/examples/example_extending_classification.html#sphx-glr-examples-example-extending-classification-py.
